### PR TITLE
fix(FEC-10312): ready promise rejected in recoverable error

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

The ready promise rejected only when a critical error occurred
this makes the player ready before it actually loads the source

Solves FEC-10312

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [x] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Comment by OrenMe** [comment]
good catch @yairans, in which use cases can this happen?
Please open internal tracking ticket as well.

**Comment by yairans** [comment]
@OrenMe I caught it in  unregistered -plugin error

File: test/src/player.spec.js [hunk 1]
```diff
@@ -209,6 +209,24 @@ describe('Player', function() {
               done();
             });
           });
+
+          it('should resolved successfully even recoverable error occurred', done => {
```
Comment (OrenMe) [inline test/src/player.spec.js:213]:
```suggestion
          it('should resolved successfully even when recoverable error occurred', done => {
```

**Comment by OrenMe** [comment]
Flow failing :-(

**Pull request merged**